components/batt/src/batt.c

```diff
@@ -0,0 +1,234 @@
+#include "batt.h"
+
+#include <ember_taskglue.h>
+#include <esp_adc/adc_cali.h>
+#include <esp_adc/adc_cali_scheme.h>
+#include <esp_adc/adc_continuous.h>
+#include <freertos/FreeRTOS.h>
+#include <freertos/semphr.h>
+#include <freertos/task.h>
+#include <hal/adc_types.h>
+#include <opencan_rx.h>
+#include <opencan_tx.h>
+
+#include <math.h>
+
+#define ADC_TASK_STACK_SIZE 2048
+#define SAMPLING_RATE_HZ    256
+#define SAMPLES		    256
+#define FRAME_SAMPLES	    128
+#define FRAME_SIZE \
+	(sizeof(adc_digi_output_data_t) * FRAME_SAMPLES * ADC_CHANNELS)
+
+#define SAMPLES_OUT_SIZE   1024	  // how to decide the size of the output
+#define PREV_SAMPLES_DELAY 0.010  // why is this the delay time
+#define PREV_SAMPLES_SIZE  (size_t)(PREV_SAMPLES_DELAY * SAMPLING_RATE_HZ)
+
+#define POOL_SIZE (FRAME_SIZE * 2)
+
+#define ADC_CONV	       ADC_CONV_SINGLE_UNIT_1
+#define ADC_DIGI_OUTPUT_FORMAT ADC_DIGI_OUTPUT_FORMAT_TYPE2
+
+#define BV_ADC_ATTEN	ADC_ATTEN_DB_12
+#define BV_ADC_UNIT	ADC_UNIT_2
+#define BV_ADC_CHANNEL	ADC_CHANNEL_7  // GPIO_8
+#define BV_ADC_BITWIDTH SOC_ADC_DIGI_MAX_BITWIDTH
+
+// #define PWM_PIN GPIO_NUM_37
+// #define PWM_FREQUENCY 1000
+// #define PWM_INIT_DUTY_CYCLE 0
+// #define PWM_RESOLUTION 10
+
+adc_continuous_handle_t handle = NULL;
```

Comment:
```suggestion
static adc_continuous_handle_t handle;
```

This will automatically be set to `NULL` if we don't initialize it as a global variable. I'd also change the handle to be `static` so that we don't pollute the global symbol table.